dexcom/fetch/runner.go

```diff
@@ -715,6 +722,15 @@ func (t *TaskRunner) storeDatumArray(datumArray data.Data) error {
 		if err := t.updateDataSourceWithDataTime(earliestDataTime, latestDataTime); err != nil {
 			return err
 		}
+
+		// Determine last known timezone offset and persist with the data set
+		var timezoneOffset *int
+		for index := endIndex - 1; index >= 0 && timezoneOffset == nil; index-- {
+			timezoneOffset = datumArray[index].GetTimeZoneOffset()
+		}
```

Comment:
Should there be a `break` here?

Reply:
The `for` loop has a `&& timezoneOffset == nil` condition to exit the loop once the first non-`nil` `timezoneOffset` is found. 

However, given your comment, it is non-obvious, therefore I'll update to make it more readable.

Reply:
Updated to make it more clear.